File: Messenger.Client/src/models/MessageEntity.ts
import { IAttachment } from "./interfaces/IAttachment";
import { IMessage } from "./interfaces/IMessage";

export default class MessageEntity implements IMessage {
  public id;
  public text;
  public isEdit;
  public ownerId;
  public ownerDisplayName;
  public ownerAvatarLink;
  public replyToMessageId;
  public replyToMessageText;
  public replyToMessageAuthorDisplayName;
  public attachments;
  public chatId;
  public dateOfCreate;

  constructor(
    id: string,
    text: string,
    isEdit: boolean,
    ownerId: string,
    ownerDisplayName: string,
    ownerAvatarLink: string | null,
    replyToMessageId: string | null,
    replyToMessageText: string | null,
    replyToMessageAuthorDisplayName: string | null,
    attachments: IAttachment[],
    chatId: string,
    dateOfCreate: string
  ) {
    this.id = id;
    this.text = text;
    this.isEdit = isEdit;
    this.ownerId = ownerId;
    this.ownerDisplayName = ownerDisplayName;
    this.ownerAvatarLink = ownerAvatarLink;
    this.replyToMessageId = replyToMessageId;
    this.replyToMessageText = replyToMessageText;
    this.replyToMessageAuthorDisplayName = replyToMessageAuthorDisplayName;
    this.attachments = attachments;
    this.chatId = chatId;
    this.dateOfCreate = dateOfCreate;
  }
}

File: Messenger.Domain/Entities/MessageEntity.cs
using FluentValidation;
using Messenger.Domain.Entities.Abstraction;
using Messenger.Domain.Entities.Validation;

namespace Messenger.Domain.Entities;

public class MessageEntity : IBaseEntity
{
	public Guid Id { get; set; } = Guid.NewGuid();

	public string Text { get; set; }
	
	public bool IsEdit { get; set; }
	
	public Guid? OwnerId { get; set; }

	public UserEntity? Owner { get; set; }

	public Guid? ReplyToMessageId { get; set; }
	
	public MessageEntity? ReplyToMessage { get; set; }
	
	public Guid ChatId { get; set; }

	public ChatEntity Chat { get; set; }

	public List<AttachmentEntity> Attachments { get; set; } = new(); 

	public DateTime DateOfCreate { get; set; } = DateTime.UtcNow;
	
	public ChatEntity? LastMessageByChat { get; set; }

	public List<DeletedMessageByUserEntity> DeletedMessageByUsers { get; set; }

	public MessageEntity(string text, Guid? ownerId, Guid? replyToMessageId, Guid chatId)
	{
		Text = text;
		OwnerId = ownerId;
		ReplyToMessageId = replyToMessageId;
		ChatId = chatId;

		new MessageEntityValidator().ValidateAndThrow(this);
	}
}